lib/galaxy/tool_util/cwl/util.py

```diff
@@ -148,12 +155,10 @@ def galactic_job_json(
     datasets = []
     dataset_collections = []
 
-    def response_to_hda(target: UploadTarget, upload_response: Dict[str, Any]) -> Dict[str, str]:
-        assert isinstance(upload_response, dict), upload_response
-        assert "outputs" in upload_response, upload_response
-        assert len(upload_response["outputs"]) > 0, upload_response
-        dataset = upload_response["outputs"][0]
-        datasets.append(dataset)
```

Comment:
This looks suspicious, without this `datasets` is returned empty but it's used in some places by the callers.